compiler/rustc_hir/src/lang_items.rs

```diff
@@ -314,6 +314,8 @@ language_item_table! {
     TryTraitFromResidual,    sym::from_residual,       from_residual_fn,           Target::Method(MethodKind::Trait { body: false });
     TryTraitFromOutput,      sym::from_output,         from_output_fn,             Target::Method(MethodKind::Trait { body: false });
     TryTraitBranch,          sym::branch,              branch_fn,                  Target::Method(MethodKind::Trait { body: false });
+    // Language items useful for MIR optimization passes
+    SliceLen,                sym::slice_len_fn,        slice_len_fn,               Target::Method(MethodKind::Inherent);
```

Comment:
Can you push this just a but further up to above `Try*`?

Reply:
Sure, just wanted to keep some notes